generate-example-index.ts

```diff
@@ -0,0 +1,660 @@
+              this._generateJSFiddle(example),
+              this._generateCodePen(example)
+            )
+        );
+
+        const image = cheerio("<a>")
+          .attr("href", example.path)
+          .append(
+            cheerio("<div>")
+              .addClass("example-image")
+              .append(
+                cheerio("<img>")
+                  .attr("src", this._pageToScreenshotPath(example.path))
+                  .attr("alt", key)
+              )
+          );
+
+        const item = cheerio("<span>")
+          .addClass("example-link")
+          .append(header, image);
+
+        list.append(item);
+
+        this._screenshotTodo.push({
+          pagePath: example.path,
+          selector: example.selector,
+          delay: example.delay
+        });
+      } else {
+        section.append(await this._processGroup(example, key, level + 1));
+      }
+    }
+
+    return section;
+  }
+  private _generatePlaygroundData(
+    example: Example
+  ): {
+    code: {
+      css: string;
+      html: string;
+      js: string;
+    };
+    resources: {
+      css: string[];
+      js: string[];
+    };
+  } {
+    // JavaScript
+    const eventListeners = (Object.entries(
+      example.$("body").get(0).attribs
+    ) as [string, string][])
+      .filter(([name]): boolean => /^on/.test(name))
+      .map(([name, value]): [string, string] => [name.slice(2), value])
+      .map(
+        ([name, value]): string =>
+          `window.addEventListener("${name}", () => { ${value} });`
+      )
+      .join("\n");
+    const js = formatJS(
+      example
+        .$("script")
+        .map((_i, elem) => elem.children[0])
+        .get()
+        .map((elem): string => elem.data)
+        .join("") +
+        "\n\n;" +
+        eventListeners
+    );
+
+    // Cascading Style Sheets
+    const css = formatCSS(
+      example
+        .$("style")
+        .map((_i, elem) => elem.children[0])
+        .get()
+        .map((elem): string => elem.data)
+        .join("")
+    );
+
+    // Hypertext Markup Language
+    const $html = cheerio.load(example.$("body").html());
+    $html("script").remove();
+
+    const html = formatHTML($html("body").html());
+
+    // Resources
+    const fixPath = (rawPath: string): string =>
+      /^https?:\/\//.test(rawPath)
+        ? rawPath
+        : path
+            .resolve(path.dirname(example.path) + path.sep + rawPath)
+            .replace(this._projectPath, this._webURL);
+    const resources = {
+      js: example
+        .$("script")
+        .map((_i, elem): string => cheerio(elem).attr("src"))
+        .get()
+        .map(fixPath),
+      css: example
+        .$("link[rel='stylesheet']")
+        .map((_i, elem): string => cheerio(elem).attr("href"))
+        .get()
+        .map(fixPath)
+    };
+
+    return {
+      code: {
+        css,
+        html,
+        js
+      },
+      resources
+    };
+  }
+  private _generateJSFiddle(example: Example): Cheerio {
```

Comment:
For me jsfiddle always runs into a timeout.

Reply:
It's not just you! jsfiddle.net is down: http://isup.me/jsfiddle.net .